pkg/controllers/provisioning/provisioner.go

```diff
@@ -131,6 +131,14 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	if len(results.NewNodeClaims) == 0 {
 		return reconcile.Result{}, nil
 	}
+	np , err := nodepoolutil.Get(ctx, p.kubeClient, results.NewNodeClaims[0].OwnerKey)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if !np.GetCondition(v1beta1.NodeClassConditionTypeReady){
+		logging.FromContext(ctx).Debugf("waiting for nodeclass to become ready")
```

Comment:
I think the readiness condition in the NodePool should be enough for observability, not sure if we need a log line here